core/src/libraries/Oracle.sol

```diff
@@ -129,6 +177,7 @@ library Oracle {
 
                 if (timeL < target && seed == observationIndex) {
                     uint56 delta = uint56(target - timeL);
+                    uint128 liquidity = pool.liquidity();
```

Comment:
This is the only if statement where `liquidity` is necessary for the computation, so instead of fetching `liquidity` every time (and rarely using it) we fetch `liquidity` only after determining that it's needed.